src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -997,66 +1026,89 @@ cpdef object compute_network_structured(
     cdef float[:,:,::1] flowveldepth = flowveldepth_nd
+    cdef float lp_outflow, lp_water_elevation
+    cdef int id = 0
     #Run time
     with nogil:
-      while timestep < nsteps:
+      while timestep < nsteps+1:
         #for r in reach_objects:
         for i in range(num_reaches):
-              r = reach_structs[i]
+              r = &reach_structs[i]
               #Need to get quc and qup
               upstream_flows = 0.0
               previous_upstream_flows = 0.0
+
               for _i in range(r._num_upstream_ids):#Explicit loop reduces some overhead
                 id = r._upstream_ids[_i]
                 upstream_flows += flowveldepth[id, timestep, 0]
                 previous_upstream_flows += flowveldepth[id, timestep-1, 0]
-              #Index of segments required to process this reach
-              #segment_ids = []#[segment.id for segment in r._segments]
 
               if assume_short_ts:
-                  upstream_flows = previous_upstream_flows
-              #Create compute reach kernel input buffer
-              #for i, segment in enumerate(r.segments):
-              for i in range(r._num_segments):
-                segment = r._segments[i]
-                buf_view[i, 0] = qlat_array[ segment.id, <int>(timestep/qlat_resample)]
-                buf_view[i, 1] = segment.dt
-                buf_view[i, 2] = segment.dx
-                buf_view[i, 3] = segment.bw
-                buf_view[i, 4] = segment.tw
-                buf_view[i, 5] = segment.twcc
-                buf_view[i, 6] = segment.n
-                buf_view[i, 7] = segment.ncc
-                buf_view[i, 8] = segment.cs
-                buf_view[i, 9] = segment.s0
-                buf_view[i, 10] = flowveldepth[segment.id, timestep-1, 0]
-                buf_view[i, 11] = 0.0 #flowveldepth[segment.id, timestep-1, 1]
-                buf_view[i, 12] = flowveldepth[segment.id, timestep-1, 2]
-
-              compute_reach_kernel(previous_upstream_flows, upstream_flows,
-                                   r._num_segments, buf_view,
-                                   out_buf,
-                                   assume_short_ts)
-              #Copy the output out
-              for i in range(r._num_segments):
-                flowveldepth[r._segments[i].id, timestep, 0] = out_buf[i, 0]
-                flowveldepth[r._segments[i].id, timestep, 1] = out_buf[i, 1]
-                flowveldepth[r._segments[i].id, timestep, 2] = out_buf[i, 2]
+                upstream_flows = previous_upstream_flows
+
+              if r.type == compute_type.RESERVOIR_LP:
+                """
+                id = r.reach.lp.lake_number
+                with gil:
+                  print("id ", r.reach.lp)
+                  print("id ", id)
+                  printf("timestep %d\n",  timestep)
+                  exit(1)
+                """
+
+
+                run(r, upstream_flows, 0.0, 300, &lp_outflow, &lp_water_elevation)
```

Comment:
should probably give this function a name that  indicates it is running a levelpool reservoir